# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/ffvpx/libavcodec/error_resilience.h
# Commit: 186e2a334f2d
# Full Hash: 186e2a334f2deb37b29f44b28bc8c83babeeb9a4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-09 09:37:17
# Regressor Bug: 1877221
# File Overlap Count: 1
# Description:
#   Bug 1877221 - Pull in new ffmpeg code, reapply patch. r=media-playback-reviewers,chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199909
# ==============================================================================

diff -r c5920de78d16 -r 186e2a334f2d media/ffvpx/libavcodec/error_resilience.h
--- a/media/ffvpx/libavcodec/error_resilience.h	Thu Feb 08 10:39:53 2024 +0000
+++ b/media/ffvpx/libavcodec/error_resilience.h	Thu Feb 08 10:39:54 2024 +0000
@@ -24,7 +24,6 @@
 
 #include "avcodec.h"
 #include "me_cmp.h"
-#include "threadframe.h"
 
 ///< current MB is the first after a resync marker
 #define VP_START               1
@@ -40,7 +39,7 @@
 
 typedef struct ERPicture {
     AVFrame *f;
-    ThreadFrame *tf;
+    const struct ThreadFrame *tf;
 
     // it is the caller's responsibility to allocate these buffers
     int16_t (*motion_val[2])[2];
@@ -90,7 +89,17 @@
 } ERContext;
 
 void ff_er_frame_start(ERContext *s);
-void ff_er_frame_end(ERContext *s);
+
+/**
+ * Indicate that a frame has finished decoding and perform error concealment
+ * in case it has been enabled and is necessary and supported.
+ *
+ * @param s                  ERContext in use
+ * @param decode_error_flags pointer where updated decode_error_flags are written
+ *                           if supplied; if not, the new flags are directly
+ *                           applied to the AVFrame whose errors are concealed
+ */
+void ff_er_frame_end(ERContext *s, int *decode_error_flags);
 void ff_er_add_slice(ERContext *s, int startx, int starty, int endx, int endy,
                      int status);
 